File: slack-poster/src/test/java/dev/number6/slackposter/dagger/RecordingHttpAdaptor.java
package dev.number6.slackposter.dagger;

import dev.number6.slack.CallResponse;
import dev.number6.slack.port.HttpPort;
import com.amazonaws.services.lambda.runtime.LambdaLogger;

import java.util.HashMap;
import java.util.Map;

public class RecordingHttpAdaptor implements HttpPort {

    private Map<String, String> posts = new HashMap<>();

    @Override
    public CallResponse get(String url, LambdaLogger logger) {
        return new CallResponse("quack");
    }

    @Override
    public CallResponse post(String url, String body, LambdaLogger logger) {

        posts.put(url, body);
        return new CallResponse("quack");
    }

    public Map<String, String> getPosts() {
        return posts;
    }
}

File: slack-poster/src/test/java/dev/number6/slackposter/dagger/RecordingHttpAdaptor.kt
package dev.number6.slackposter.dagger

import com.amazonaws.services.lambda.runtime.LambdaLogger
import dev.number6.slack.CallResponse
import dev.number6.slack.port.HttpPort
import java.util.*

class RecordingHttpAdaptor : HttpPort {
    private val posts: MutableMap<String?, String?> = HashMap()
    override fun get(url: String, logger: LambdaLogger): CallResponse {
        return CallResponse("quack")
    }

    override fun post(url: String, body: String, logger: LambdaLogger): CallResponse {
        posts[url] = body
        return CallResponse("quack")
    }

    fun getPosts(): Map<String?, String?> {
        return posts
    }
}